src/features/emojiMod.ts

```diff
@@ -16,45 +12,62 @@ const config = {
   deletionThreshold: Infinity
 };
 
-const warningMessages = {};
+type WarningMessages = {
+  [messageId: string]: Message;
+};
+
+const warningMessages: WarningMessages = {};
 
 const thumbsDownEmojis = ["👎", "👎🏻", "👎🏼", "👎🏽", "👎🏾", "👎🏿"];
 
-const reactionHandlers = {
-  "⚠️": (bot, reaction, message, member) => {
-    if (!isStaff(member)) {
-      return;
-    }
-    if (isStaff(message.guild.members.cache.get(message.author.id))) {
+type ReactionHandlers = {
+  [emoji: string]: (
+    reaction: MessageReaction,
+    message: Message,
+    member: GuildMember
+  ) => void;
+};
+
+const reactionHandlers: ReactionHandlers = {
+  "⚠️": (reaction, message, member) => {
+    // Skip if the user that reacted isn't in the staff of the post is from someone
+    // from the staff
+    if (
+      !message.guild ||
+      !message.author ||
+      !isStaff(member) ||
+      isStaff(message.guild.member(message.author.id))
```

Comment:
Why are we removing the usages of `.cache` here, among the other places too?

Reply:
`message.guild.member()` fetches the member directly, so we aren't accessing the cache